src/components/Movies/MovieFilterBar.tsx

```diff
@@ -1,14 +1,32 @@
-import React, { FC } from 'react'
+import React, { FC, MouseEventHandler } from 'react'
 import classNames from "classnames"
 
 import styles from './MovieFilterBar.scss'
 
+import {SortOrderType} from "../../App";
+
 const genres = ['All','Documentary','Comedy','Horror','Crime']
 
-const MovieFilterBar: FC = () => {
+interface MovieFilterBarProps {
+  sortOrder: SortOrderType
+  onSortOrderChange: (k: SortOrderType) => void
+}
+
+const MovieFilterBar: FC<MovieFilterBarProps> = ({
+  sortOrder,
+  onSortOrderChange
+}) => {
   const activeGenre = 'All'
   const activeSort = 'Release Date'
 
+  const invertSortOrder = () => {
+    if (sortOrder === 'asc') {
+      onSortOrderChange('desc')
```

Comment:
I suspect this one will cause DOUBLE sorting. Please check how many times your `handleSortOrderChange` is called WHEN you change the sort order.

Reply:
I have checked and double sorting was not detected.